modules/mesh-layers/src/scenegraph-layer/scenegraph-layer.js

```diff
@@ -19,69 +19,19 @@
 // THE SOFTWARE.
 
 import {Layer} from '@deck.gl/core';
-import {fp64} from '@luma.gl/core';
-import {createGLTFObjects} from '@luma.gl/addons';
+import {fp64, ScenegraphNode, log} from '@luma.gl/core';
 import {getMatrixAttributes} from '../utils/matrix';
 
-const {fp64LowPart} = fp64;
+import vs from './scenegraph-layer-vertex.glsl';
+import fs from './scenegraph-layer-fragment.glsl';
 
-const vs = `
-  // Instance attributes
-  attribute vec3 instancePositions;
-  attribute vec2 instancePositions64xy;
-  attribute vec4 instanceColors;
-  attribute vec3 instancePickingColors;
-  attribute mat3 instanceModelMatrix;
-  attribute vec3 instanceTranslation;
-
-  // Uniforms
-  uniform float sizeScale;
-
-  // Attributes
-  attribute vec4 POSITION;
-
-  #ifdef HAS_UV
-    attribute vec2 TEXCOORD_0;
-    varying vec2 vTEXCOORD_0;
-  #endif
-  varying vec4 vColor;
-
-  void main(void) {
-    #ifdef HAS_UV
-      vTEXCOORD_0 = TEXCOORD_0;
-    #endif
-    vColor = instanceColors;
-
-    vec3 pos = (instanceModelMatrix * POSITION.xyz) * sizeScale + instanceTranslation;
-    pos = project_size(pos);
-
-    vec4 position_commonspace;
-    gl_Position = project_position_to_clipspace(instancePositions, instancePositions64xy, pos, position_commonspace);
-    picking_setPickingColor(instancePickingColors);
-  }
-`;
-
-const fs = `
-  #ifdef HAS_UV
-    varying vec2 vTEXCOORD_0;
-    uniform sampler2D u_BaseColorSampler;
-  #endif
-  varying vec4 vColor;
-
-  void main(void) {
-    #ifdef HAS_UV
-      gl_FragColor = (vColor / 255.) * texture2D(u_BaseColorSampler, vTEXCOORD_0);
-    #else
-      gl_FragColor = vColor / 255.;
-    #endif
-
-    gl_FragColor = picking_filterPickingColor(gl_FragColor);
-  }
-`;
+const {fp64LowPart} = fp64;
 
 const DEFAULT_COLOR = [255, 255, 255, 255];
 
 const defaultProps = {
+  scenegraph: {type: 'object', value: null, async: true},
+
   sizeScale: {type: 'number', value: 1, min: 0},
   getPosition: {type: 'accessor', value: x => x.position},
   getColor: {type: 'accessor', value: x => x.color || DEFAULT_COLOR},
```

Comment:
Use constant as default value

Reply:
What constant?

Reply:
`value: DEFAULT_COLOR`. This is more efficient than a callback (`value: d => d.color`)

Reply:
Changing this completely breaks everything for somer reason... Staying up late to investigate.

Reply:
Let's push it off to another PR.

Reply:
ok will add todo